examples/tut_dot-product.cpp

```diff
@@ -179,6 +188,29 @@ int main(int RAJA_UNUSED_ARG(argc), char **RAJA_UNUSED_ARG(argv[]))
 
 //----------------------------------------------------------------------------//
 
+#if defined(RAJA_ENABLE_SYCL)
+  std::cout << "\n Running RAJA SYCL dot product...\n";
+
+  // _rajasycl_dotprod_start
+  RAJA::ReduceSum<RAJA::sycl_reduce, double> hpdot(0.0);
+
+  RAJA::forall<RAJA::sycl_exec_nontrivial<SYCL_BLOCK_SIZE, false>>(RAJA::RangeSegment(0, N),
```

Comment:
When is `sycl_exec_nontrivial` needed? is it just reductions?

Reply:
They are needed for any kernel that contains anything that is not-trivially copy-able. @homerdin added a note to the policy section of the user guide about this. We are encouraging users to start with the "nontrivial" versions of the policies to make sure their code works. I believe @homerdin was exploring some C++ std machinery to see if this can be made simpler for users.